### PR TITLE
Updated the Workflow guide to remove the usage of the deprecated meth…

### PR DESCRIPTION
**Pull request by sagarkapare**
…od and also added unit test case

### CODE REVIEW

**Comment by sagarkapare** [comment]
@bdmogal Please help review. Thanks!

File: src/main/java/co/cask/cdap/guides/workflow/PurchaseEventVerifier.java [hunk 1]
```diff
@@ -22,21 +21,22 @@ public boolean apply(WorkflowContext workflowContext) {
...
+    Value mapInputRecords = token.get(taskCounterGroupName + "." + mapInputRecordsCounterName,
+                                      WorkflowToken.Scope.SYSTEM);
 
-    if (taskCounter.containsKey("MAP_INPUT_RECORDS")) {
-      long mapInputRecordNumber = taskCounter.get("MAP_INPUT_RECORDS");
-      long mapOutputRecordNumber = taskCounter.get("MAP_OUTPUT_RECORDS");
-      // Return true if at least 80% of the records were successfully parsed and emitted
-      // by previous map job
-      return (mapOutputRecordNumber >= (mapInputRecordNumber * 80/100));
+    String mapOutputRecordsCounterName = "MAP_OUTPUT_RECORDS";
```
Comment (bdmogal) [inline src/main/java/co/cask/cdap/guides/workflow/PurchaseEventVerifier.java:30]: is this variable necessary? can just use the literal in the next line?

File: README.rst [hunk 2]
```diff
@@ -207,22 +207,23 @@ Lets take a closer look at the predicate ``PurchaseEventVerifier``.
...
+      Value mapInputRecords = token.get(taskCounterGroupName + "." + mapInputRecordsCounterName,
+                                        WorkflowToken.Scope.SYSTEM);
 
-      if (taskCounter.containsKey("MAP_INPUT_RECORDS")) {
-        long mapInputRecordNumber = taskCounter.get("MAP_INPUT_RECORDS");
-        long mapOutputRecordNumber = taskCounter.get("MAP_OUTPUT_RECORDS");
-        // Return true if at least 80% of the records were successfully parsed and emitted
-        // by previous map job
-        return (mapOutputRecordNumber >= (mapInputRecordNumber * 80/100));
+      String mapOutputRecordsCounterName = "MAP_OUTPUT_RECORDS";
```
Comment (bdmogal) [inline README.rst:216]: ditto

**Comment by bdmogal** [comment]
couple minor comments. Should also update the CDAP dependency to 3.1.0 (from 3.1.0-SNAPSHOT).

Also, do we need any README updates?

File: src/main/java/co/cask/cdap/guides/workflow/PurchaseEventVerifier.java (hunk 1 above)
Comment (sagarkapare) [inline src/main/java/co/cask/cdap/guides/workflow/PurchaseEventVerifier.java:30]: We can, however was explicitly using the variables here to highlight the counter group and name are separated by `.` character.

**Comment by sagarkapare** [comment]
@bdmogal Updated the pom.xml and README. Please take a look again. Thanks!

**Comment by bdmogal** [comment]
:+1:

**Pull request merged**